### PR TITLE
Merge sort

### PR DESCRIPTION
**Pull request by TheQuangNguyen**

### CODE REVIEW

File: Data-Structures/src/main/java/sorting/MergeSort.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
+package sorting;
+
+public class MergeSort implements SortingAlgorithm {
+
+    // array to store values of original array to enable in place sorting of original array
+    private int[] aux;
+    // Sort array by recursively merging two sorted arrays together
+    @Override
+    public int[] sort(int[] arr) {
```
Comment (jeffborda) [inline Data-Structures/src/main/java/sorting/MergeSort.java]: You may want to consider making your sort a `static` method, that way you don't have to instantiate`MergeSort` objects to use it.

File: Data-Structures/src/test/java/sorting/MergeSortTest.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+import static org.junit.Assert.*;
+
+public class MergeSortTest {
+
+    MergeSort mergeSort;
+    @Before
+    public void setUp() throws Exception {
+        mergeSort = new MergeSort();
+    }
+
```
Comment (jeffborda) [inline Data-Structures/src/test/java/sorting/MergeSortTest.java]: Excellent test cases... you've got your edge cases covered.
